File: 10340 - All in All/10340 - All in All.py
from sys import stdin as y


def main():
    line = y.readline().strip()
    while line:
        s, t = line.split()
        sizes, sizet = len(s), len(t)
        ps = 0
        for i in range(sizet):
            if s[ps] == t[i]: ps += 1
        print("Yes") if ps == sizes else print("No")
        line = y.readline().strip()


main()

File: 10340 - All in All/10340 - All in All.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
    //freopen("out.txt","w",stdout);
    string s, t;
    int sizes, sizet, ps;
    while (cin >> s >> t){
        sizes = s.size();
        sizet = t.size();
        ps = 0;
        for (int i = 0; i < sizet; i++){
            if (s[ps] == t[i]) {ps ++;}
        }
        if (ps == sizes) {printf("Yes\n");}
        else{printf("No\n");}
    }
    return 0;
}
